src/alda/repl/AldaRepl.java

```diff
@@ -144,18 +148,11 @@ public void run() {
           history.append("\n");
 
           // If we're good, we should check to see if we reset the instrument
-		  if (playResponse != null &&
-			  playResponse.score != null &&
-			  playResponse.score.currentInstruments.length > 0) {
-			  // If we have multiple instruments, pick the first one.
-			  String instrument = playResponse.score.currentInstruments[0];
-			  if (instrument.length() > 0) {
-				  // Pick the first letter of the instrument
-				  promptPrefix = Character.toString(instrument.charAt(0));
-			  } else {
-				  promptPrefix = "";
-			  }
-		  }
+          if (playResponse != null && playResponse.currentInstrument() != null) {
+            // If we have multiple instruments, pick the first one.
+            String instrument = playResponse.currentInstrument();
+            this.setPromptPrefix(instrument);
```

Comment:
The Clojure implementation is here, for reference: https://github.com/alda-lang/alda-repl-clj/blob/master/src/alda/repl/core.clj#L60-L75